src/dmd/target.d

```diff
@@ -1451,7 +1451,7 @@ struct TargetCPP
         case DigitalMars: return predef("CppRuntime_DigitalMars");
         case Gcc:         return predef("CppRuntime_Gcc");
         case Microsoft:   return predef("CppRuntime_Microsoft");
-        case Sun:         return predef("CppRuntime_Sun");
+        case Sun:         return predef("CppRuntime_Gcc");
```

Comment:
You don't want to change this, rather make the Solaris runtime default to Gcc.

https://github.com/dlang/dmd/blob/28d77d18c79ad0b7b5fcdb8398c1fa0074df1f43/src/dmd/target.d#L1308-L1309